test/Microsoft.Azure.WebJobs.Host.FunctionalTests/Queues/QueueListenerTests.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+
+using System;
+using System.Diagnostics;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Queues;
+using Microsoft.Azure.WebJobs.Host.Queues.Listeners;
+using Microsoft.Azure.WebJobs.Host.Storage.Queue;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.FunctionalTests.Queues
+{
+    public class QueueListenerTests : IClassFixture<QueueListenerTests.TestFixture>
+    {
+        public QueueListenerTests(TestFixture fixture)
+        {
+            Fixture = fixture;
+        }
+
+        public TestFixture Fixture { get; set; }
+
+        [Fact]
+        public async Task UpdatedQueueMessage_RetainsOriginalProperties()
```

Comment:
A big test that makes sure we pass the original values to each overloaded method.